component/azstorage/block_blob.go

```diff
@@ -854,15 +882,14 @@ func trackUpload(name string, bytesTransferred int64, count int64, uploadPtr *in
 }
 
 // WriteFromFile : Upload local file to blob
-func (bb *BlockBlob) WriteFromFile(name string, metadata map[string]string, fi *os.File) (err error) {
+func (bb *BlockBlob) WriteFromFile(name string, metadata map[string]*string, fi *os.File) (err error) {
 	log.Trace("BlockBlob::WriteFromFile : name %s", name)
 	//defer exectime.StatTimeCurrentBlock("WriteFromFile::WriteFromFile")()
 
-	blobURL := bb.Container.NewBlockBlobURL(filepath.Join(bb.Config.prefixPath, name))
+	blobClient := bb.Container.NewBlockBlobClient(filepath.Join(bb.Config.prefixPath, name))
 	defer log.TimeTrack(time.Now(), "BlockBlob::WriteFromFile", name)
 
-	var uploadPtr *int64 = new(int64)
-	*uploadPtr = 1
+	uploadPtr := to.Ptr(int64(1))
```

Comment:
where is this used? Do we reallly need this variable.

Reply:
This is used for health monitor.